### ISSUE TITLE
passwordInvalid is not a valid attribute after upgrading to version 14

### ISSUE THREAD

**Issue by kolaente** [nextcloud/server#12750]
### Steps to reproduce
Upgrade from version 12 to 14 through 13, then after successfully upgraded to version 14, try to login.

### Expected behaviour
Login should work without issues.

### Actual behaviour
You are greeted with a screen which says "Internal server error - passwordInvalid is not a valid attribute" - like in https://help.nextcloud.com/t/after-downgrade-from-15b2-14-0-3/41469

## Server configuration detail

**Operating system:** Linux 4.14.0-0.bpo.2-amd64 #1 SMP Debian 4.14.7-1~bpo9+1 (2017-12-22) x86_64

**Webserver:** nginx/1.15.7 (fpm-fcgi)

**Database:** mariadb 10.3.11

**PHP version:** 

7.2.12
Modules loaded: Core, date, libxml, openssl, pcre, sqlite3, zlib, ctype, curl, dom, fileinfo, filter, ftp, hash, iconv, json, mbstring, SPL, PDO, session, posix, readline, Reflection, standard, SimpleXML, pdo_sqlite, Phar, tokenizer, xml, xmlreader, xmlwriter, mysqlnd, cgi-fcgi, apcu, exif, gd, intl, ldap, memcached, pcntl, pdo_mysql, pdo_pgsql, redis, sodium, zip, Zend OPcache

**Nextcloud version:** 14.0.4 - 14.0.4.2

**Updated from an older Nextcloud/ownCloud or fresh install:** 

**Where did you install Nextcloud from:** docker

<details><summary>Signing status</summary>

</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - accessibility: 1.0.1
 - activity: 2.7.0
 - admin_audit: 1.4.0
 - calendar: 1.6.4
 - cloud_federation_api: 0.0.1
 - comments: 1.4.0
 - contacts: 2.1.7
 - dav: 1.6.0
 - deck: 0.5.0
 - federatedfilesharing: 1.4.0
 - federation: 1.4.0
 - files: 1.9.0
 - files_pdfviewer: 1.3.2
 - files_sharing: 1.6.2
 - files_texteditor: 2.6.0
 - files_trashbin: 1.4.1
 - files_versions: 1.7.1
 - files_videoplayer: 1.3.0
 - firstrunwizard: 2.3.0
 - gallery: 18.1.0
 - issuetemplate: 0.4.0
 - logreader: 2.0.0
 - lookup_server_connector: 1.2.0
 - mail: 0.11.0
 - notifications: 2.2.1
 - oauth2: 1.2.1
 - password_policy: 1.4.0
 - provisioning_api: 1.4.0
 - serverinfo: 1.4.0
 - sharebymail: 1.4.0
 - support: 1.0.0
 - systemtags: 1.4.0
 - theming: 1.5.0
 - twofactor_backupcodes: 1.3.1
 - user_external: 0.4
 - workflowengine: 1.4.0
Disabled:
 - encryption
 - files_external
 - nextcloud_announcements
 - survey_client
 - updatenotification
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "apps_paths": [
        {
            "path": "\/var\/www\/html\/apps",
            "url": "\/apps",
            "writable": false
        },
        {
            "path": "\/var\/www\/html\/custom_apps",
            "url": "\/custom_apps",
            "writable": true
        }
    ],
    "memcache.local": "\\OC\\Memcache\\APCu",
    "memcache.distributed": "\\OC\\Memcache\\Redis",
    "memcache.locking": "\\OC\\Memcache\\Redis",
    "redis": {
        "host": "***REMOVED SENSITIVE VALUE***",
        "port": 6379
    },
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "dbtype": "mysql",
    "version": "14.0.4.2",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "mail_from_address": "***REMOVED SENSITIVE VALUE***",
    "logtimezone": "UTC",
    "logdateformat": "Y-m-d H:i:s",
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "installed": true,
    "mail_smtpmode": "smtp",
    "mail_domain": "***REMOVED SENSITIVE VALUE***",
    "maintenance": false,
    "updater.secret": "***REMOVED SENSITIVE VALUE***",
    "theme": "",
    "loglevel": 2,
    "mail_smtpauthtype": "LOGIN",
    "mail_smtpauth": 1,
    "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
    "debug": true
}
```
</details>

**Are you using external storage, if yes which one:** 

**Are you using encryption:** false

**Are you using an external user-backend, if yes which one:** 

## Client configuration

**Browser:** Mozilla/5.0 (X11; Linux x86_64; rv:64.0) Gecko/20100101 Firefox/64.0

**Operating system:** 

## Logs

<details><summary>Nextcloud log</summary>

```
{
  "reqId": "Cc1curQ5wNAsvczFAoBq",
  "level": 3,
  "time": "2018-11-29 21:37:22",
  "remoteAddr": "10.42.144.247",
  "user": "konrad",
  "app": "index",
  "method": "GET",
  "url": "\\/apps\\/files\\/",
  "message": {
    "Exception": "BadFunctionCallException",
    "Message": "passwordInvalid is not a valid attribute",
    "Code": 0,
    "Trace": [
      {
        "file": "\\/var\\/www\\/html\\/lib\\/public\\/AppFramework\\/Db\\/Entity.php",
        "line": 146,
        "function": "setter",
        "class": "OCP\\AppFramework\\Db\\Entity",
        "type": "->",
        "args": [
          "passwordInvalid",
          [
            "0"
          ]
        ]
      },
      {
        "file": "\\/var\\/www\\/html\\/lib\\/public\\/AppFramework\\/Db\\/Entity.php",
        "line": 70,
        "function": "__call",
        "class": "OCP\\AppFramework\\Db\\Entity",
        "type": "->",
        "args": [
          "setPasswordInvalid",
          [
            "0"
          ]
        ]
      },
      {
        "file": "\\/var\\/www\\/html\\/lib\\/private\\/Authentication\\/Token\\/PublicKeyTokenMapper.php",
        "line": 86,
        "function": "fromRow",
        "class": "OCP\\AppFramework\\Db\\Entity",
        "type": "::",
        "args": [
          {
            "id": "12858",
            "uid": "konrad",
            "login_name": "konrad",
            "password": "qMYw4znQVxvsD2QjtUfTdsZcZdSGjEX4OMW1RJdEUOSNhpHAJDIDxu893YMIOQrVFlGUPP126jkaq5lRxtOJxRYx\\/3AMswBEjdXZc8O3VTdnGH\\/OgREFzv3IiJLEcPBpsVgp3QEk+TBVXCPpOwaBEWRhQ+4rNLPvhzIuv1AktXVF9IfUX7bjQdDZfWskK+VO0rlyJVKYJVniYnDxUy88jebgYjjtvgUYfTxiWeYaQsq\\/bk3HTPFB2SAznvgEsT8SkRqOaOKjUcZJ8jOQBjKXHr\\/12Dml3m9AZtsk6cTvQ8jOJp0alZJotNNk0sAJjV1nLXiF30mTiCikxrNT6ntzKw==",
            "name": "Mozilla\\/5.0 (X11; Linux x86_64; rv:64.0) Gecko\\/20100101 Firefox\\/64.0",
            "token": "*** sensitive parameter replaced ***",
            "type": "0",
            "remember": "1",
            "last_activity": "1543526178",
            "last_check": "1543526178",
            "scope": null,
            "expires": null,
            "private_key": "a3e2dd7ae1a35fd2c27e83848703ee24d641304b4bf8c0c8bfb1c43320dde9cde92d32feac7255d9bcf479479c8c305b3e4a9ed2d7b0bb574695c040e4daa5b1a293195210366044203096fea8cfccbfb2d94ac3cf18543781b9fc69d817fc0b78d06c4b84f11f7f400a7b373051c309ed77fe4a3d5751ae730c25f5193318509dfdd7a56b0d46aef054204a3b811f12c611165935a6425195c7cb33256dab1717634f5e7f92bffe7c320f401b03b054757f69f2ce432a9a2dbbddd70b1032ffc3dab712a17f2f8be6910a94eb70beaa0e15b0e089aa76f014a5ab0d2163daa5e4441d4a2e452cdb663887643f2414cf39c6fa8b5f085ef9c0b4bd7181b1f1d62f6e91d499d0010039728e1966b587f9502542dbb6253b3348826e0e45e250fd242f1124e39dc034577f3cc95c6853974c783b96c2a2c70ff1bbd13928d64da529532d528e6ad16b146ab1f7ebb0f9394fad1087a9f70989ada6ac10da6cbda1ccc5adf51b003e1e36502c83a88f188afbbfe259d9a2e5765de888604c8ca9f5382481dbd3b0eea1a430cacb22091e8ebda19364916a9d00feefec58f68958597eb2d27d7f89f3f1d74b6c634421d9b7f0c11f42bd88d23e1b45badb67e4dd20e59685b64ea5a33aa8dbfa4a93a3362c62ed57dd232b476bf86eb8009e89ffdb9da9f9783a216a3d857ca65bbcb0115d019e94bc53f23a05576bc8fdc21aeca910320b879b9cd11f073b292bac1fda9f3cd6a92f9bac239c6de1e1f25ac45a7ed0c1a05abf270bad6e87e6008f89b017039671c8e5d559e236114a47010aa77922c8371f544795e85ad9d1a3b7b5ec5a8d71c7eb2c1d9c37808178788c2cd01c46be7f4ac618e3507412f34943dd092080cdc43efe8fabe443265a582786ae51d5b16ecdb8237e88b6a689afa540069ce4cce124532d66ca5d5b67b11b8f1e8845859977af320e440b7b74dc802ea46b27dc914c6fae444e9ae6595c418c07ddc5d24a68642b7c360b06e9b48730fa6eac1abbec830f4ab0f69f0dd63da5fda775b388316d17e123e0bbf5a5649fe22b0aa0938bf2e0298dc2dc6cdd07691a36e4d25ca70741b6ae3619dc7f2d657472a97dfcae2b5a19e15b8bca6cb54b3f98b32c523b53f5aa38a9cd03b7da70a37c97db5ac5193e99ccc64d57f026f5a7a92212266b479d965870d8e8656de952a6c61cecba6db38ad8a50e8c6238bdfaf2f854263081f281e852678010d8c2f23b5b6d9b35b366631b66aba104055f133053aca08061304ab56c3a58d1224ead9d26fa30250bcac02d770766593b7e656c65638345a7dfbae4c9f9918c22b598692034bc46d919b0d91f05f75e75ffc471ecc62ac0dafcb1b2c93a59a361ee76f9faaddf84b3201f071854a7106025ebb868fbb770a2db706f9db2fbad1b9cfc5b15073e330f13552bdc5f712eab0ed81d79abc0d33cfe20382b9e5cdf58f7307454bed4a4ab91134affaf9007fffcdd6c9e562b908aeb6f783af262cfff4e452bf24f9ec9b60d11eb1c0c81f4381b6b30714c9cfb96ad557b59a29f8631cf01cec0c8d2d9073e1787eb530b6596e21c795c60361c8df325ed1955b9d114db24255a04fb17a23d23e4f91a3c152a8accf68fe1116cb1140c0ab8c1cbda933990111ea3c819cd6481a2ec1c86540065c04b061e9610b1fea4809631a3053897b9ec62e8d8f4b720d817cf8d5d809507786373b9b9baaa9e3dbb4410d8830545604d0cbf31b5d223965720703069abb34de8bec3f9ea2e22e79a6333dbee7b5ce2e3b7a8e46122941e72d236b1caaab3b605043c760f0a598ba08051a97a7e25dbc4bd92406ce4412dda4497d13afc3e7d278faad2329b903640779bcac18f695828695a9fb6d949962889f6cb484f7e3dec5ef2fb4b360cf8b2ff64a07a01e9a0bfe5ee785d03953059fdc05da9d975356973398576df4f292cc51050ffb156a773be3f31e95437c8c7346b0f271cc98ffad56f354809cf6d1b47c5391fdf14d675150c57a9adace00a765e782995da150bb4eb8641f88674fb849fc7f1c58b49018ab2277a95542b7a5d33718dd96d8955df771f4cc918ee32034e45124a7ba1918cf23d47a552583307de7a727650219d57ae05c16b39fa03e868e9a1876a9a30cc058025133c90dc8fafe124f3523a1aa68ab8d3bb25fc23ec832193bcff65e1644287477ad4975b403865b1775376bad07920c1b91c65830190f164a627e828e17a2b27da24e63141b268a439f28c583a124ed2baf8b5c2be9e3570c567cbbafdac153febc931cc18eff33cca0f080418a6a84f856f473db565a9b53c63f2edd258a59515c46384f5d6d7ce78522ee99c7ec3c7d7adf4bb60252bca20cfe84a70965f857ad18c26850580c32c76d1253ad111ab7156ad042082c4d450626106|IEEjoUfcBbs0Vron|71b9639531c3931b358fd33a05c630915677ca8abd75e949e1460bd8a46dc5e075effd6ac058dd65dabacaa5e82815d91b62ac89b0de3a2ef0cae32208d9305f",
            "public_key": "-----BEGIN PUBLIC KEY-----\nMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAuiZ7GRPib5RDmODGOn99\nrVcKoV8vpWYv+f4Gpp82K13yJI0zqW7fUUfdIObjkw4GzOtcTEASTPLn+V1P3Grl\ncTmylQalMt9Nof60gRtzWDHJrhTZMyxMOhQkVyocaQ0iaTpWvf7medk7+BBfqjDK\nrBymxK6RD9z2Z6mwSLtsMboCb2mjn0XKLHxo1veTvrTFvicvBYnjO+fB2tGnNuNM\nH7R7A7K6U3UD6P6hGDChkwo00Z7KO9faRZiYCi+qrNM63GPHFwMklKjuNtDCGE64\nqpgbD\\/wQVLgyB6MPGIn\\/VR1Pd\\/kBHmajRWXscOmSFBNuxmAcTr\\/QWxRvo62pnEkK\n1wIDAQAB\n-----END PUBLIC KEY-----\n",
            "version": "2",
            "password_invalid": "0"
          }
        ]
      },
      {
        "file": "\\/var\\/www\\/html\\/lib\\/private\\/Authentication\\/Token\\/PublicKeyTokenProvider.php",
        "line": 79,
        "function": "getToken",
        "class": "OC\\Authentication\\Token\\PublicKeyTokenMapper",
        "type": "->",
        "args": [
          "*** sensitive parameters replaced ***"
        ]
      },
      {
        "file": "\\/var\\/www\\/html\\/lib\\/private\\/Authentication\\/Token\\/Manager.php",
        "line": 115,
        "function": "getToken",
        "class": "OC\\Authentication\\Token\\PublicKeyTokenProvider",
        "type": "->",
        "args": [
          "*** sensitive parameters replaced ***"
        ]
      },
      {
        "file": "\\/var\\/www\\/html\\/lib\\/private\\/User\\/Session.php",
        "line": 728,
        "function": "getToken",
        "class": "OC\\Authentication\\Token\\Manager",
        "type": "->",
        "args": [
          "*** sensitive parameters replaced ***"
        ]
      },
      {
        "file": "\\/var\\/www\\/html\\/lib\\/private\\/User\\/Session.php",
        "line": 260,
        "function": "validateToken",
        "class": "OC\\User\\Session",
        "type": "->",
        "args": [
          "*** sensitive parameters replaced ***"
        ]
      },
      {
        "file": "\\/var\\/www\\/html\\/lib\\/private\\/User\\/Session.php",
        "line": 235,
        "function": "validateSession",
        "class": "OC\\User\\Session",
        "type": "->",
        "args": []
      },
      {
        "file": "\\/var\\/www\\/html\\/lib\\/private\\/legacy\\/app.php",
        "line": 346,
        "function": "getUser",
        "class": "OC\\User\\Session",
        "type": "->",
        "args": []
      },
      {
        "file": "\\/var\\/www\\/html\\/lib\\/private\\/legacy\\/app.php",
        "line": 113,
        "function": "getEnabledApps",
        "class": "OC_App",
        "type": "::",
        "args": []
      },
      {
        "file": "\\/var\\/www\\/html\\/lib\\/private\\/legacy\\/user.php",
        "line": 126,
        "function": "loadApps",
        "class": "OC_App",
        "type": "::",
        "args": [
          [
            "prelogin"
          ]
        ]
      },
      {
        "file": "\\/var\\/www\\/html\\/lib\\/base.php",
        "line": 721,
        "function": "setupBackends",
        "class": "OC_User",
        "type": "::",
        "args": []
      },
      {
        "file": "\\/var\\/www\\/html\\/lib\\/base.php",
        "line": 1068,
        "function": "init",
        "class": "OC",
        "type": "::",
        "args": []
      },
      {
        "file": "\\/var\\/www\\/html\\/index.php",
        "line": 40,
        "args": [
          "\\/var\\/www\\/html\\/lib\\/base.php"
        ],
        "function": "require_once"
      }
    ],
    "File": "\\/var\\/www\\/html\\/lib\\/public\\/AppFramework\\/Db\\/Entity.php",
    "Line": 115,
    "CustomMessage": "--"
  },
  "userAgent": "Mozilla\\/5.0 (X11; Linux x86_64; rv:64.0) Gecko\\/20100101 Firefox\\/64.0",
  "version": "14.0.4.2"
}
```
</details>

**Comment by kolaente** [comment]
I could solve the issue by replacing `lib/private/Authentication/Token/PublicKeyToken.php` with the one under https://github.com/nextcloud/server/blob/6b730b4c478bc4f55a89fd7d6a7c2715e2e5b829/lib/private/Authentication/Token/PublicKeyToken.php, that helped in my case. Wired though it is missing.

**Comment by ChristophWurst** [comment]
cc @rullzer :)

I assume this is resolved now -> closing

**Comment by mtapoja** [comment]
Unbelievable, but I just fixed what seemed to be a broken upgrade to 14.0.14.1 to work again with replacing **PublicKeyToken.php** and so far everything seems to work! Vielen danke, @kolaente :)
